studsapp/containers/eventListView/eventListView.js

```diff
@@ -2,26 +2,163 @@ import React from 'react';
 import { 
     StyleSheet, 
     Dimensions,
+    View,
+    Text,
+    FlatList,
+    TouchableHighlight,
+    Image,
+    ActivityIndicator
 } from 'react-native';
 import LinearGradient from 'react-native-linear-gradient';
+import { status } from 'studsapp/store/constants';
+import Icon from 'react-native-vector-icons/Ionicons';
+
+const imageSource = 'studsapp/static/images/logo.png';
 
 class EventListView extends React.Component {
-    //TODO: Actual stuff
+    constructor(props) {
+        super(props);
+        this.state = {
+            errorMessage: '',
+            listData: []
+        };
+    }
+
+    componentDidMount() {
+        this.props.getEvents();
+    }
+
+    getEventsToList = () => {
+        let eventList = [];
+        let events = this.props.events.data.events;
+
+        for(let i = 0; i < events.length; ++i) {
+            let name = events[i].companyName;
+            let date = events[i].date;
+            eventList.push({key: i.toString(), name, date});
+        }
+
+        return eventList;
+    };
+
+    componentDidUpdate(prevProps) {
+        if (this.props !== prevProps) {
+            switch (this.props.events.status) {
+                case status.LOADING:
+                    this.setState({ errorMessage: '', listData: [] });
+                    break;
+                case status.SUCCESS:
+                    let events = this.getEventsToList();
+                    this.setState({ errorMessage: '', listData: events })
+                    break;
+                case status.ERROR:
+                    this.setState({ errorMessage: this.props.events.error, listData: [] });
```

Comment:
Det här känns onödigt, du ska inte behöva kopiera över props till state. När du kör redux ska man undvika state så mkt som möjligt

Reply:
Fixed in event list view. A bigger refactor is required for the login view, so that can be fixed after the MVP is done.